Gutenpack: Enable local Jetpack Gutenberg by default

sirreal opened the pull request:
Depends on #10279

With Gutenberg blocks available, let's enable them!

#### Changes proposed in this Pull Request:

* Enable local Gutenpack builds by default

#### Testing instructions:

* Depends on #10279
* Run this branch with Gutenberg activated. Blocks should be available without needing to set any filters.

#### Proposed changelog entry for your changes:

* Enable Jetpack Gutenberg blocks by default

sirreal committed:
Enable local Jetpack Gutenberg by default

kraftbj commented on class.jetpack.php:
```diff
@@ -7312,7 +7312,7 @@ public static function load_jetpack_gutenberg() {
 			return;
 		}
-		if ( ! Jetpack::is_gutenberg_available() || ! apply_filters( 'jetpack_gutenberg', false ) ) {
+		if ( ! Jetpack::is_gutenberg_available() || ! apply_filters( 'jetpack_gutenberg', true ) ) {
```
Looks like we missed it earlier, but should add a docblock for this filter.

sirreal replied:
Good spot, both of you. Positioned and updated in 37b329f00dda4f255357944e2cb86d853b2960aa.

simison commented:
Let's remove this note while at it:

https://github.com/Automattic/jetpack/blob/7e3c013f94ef2da274ae463043cd39a99f072bda/class.jetpack.php#L7277-L7280

sirreal committed:
Update jetpack_gutenberg filter docblock

Move the docblock to the correct position.
Update to the new default value `true`.

sirreal committed:
Update jetpack_gutenberg filter documentation

sirreal commented:
> Let's remove this note while at it:

Another good spot @simison, updated that documentation in 45381e219352dfa307073a914a8390427c0343ca

This is ready for review.

ockham commented:
Should we also remove item 2. from [the doc](https://github.com/Automattic/jetpack/blob/5c76f35a2604ba157c8728527b6b7a6f5ee2ae6f/docs/guides/gutenberg-blocks.md) now?

ockham commented:
(Well, the `add_filter()` part anyway)

ockham reviewed (dismissed):
Looks good, and tests well. Left one [note](https://github.com/Automattic/jetpack/pull/10292#issuecomment-428861319), but consider it non-blocking -- we can update docs in a subsequent iteration.

sirreal committed:
Update docs to reflect default filter values

ockham commented on docs/guides/gutenberg-blocks.md:
````diff
@@ -10,10 +10,9 @@ _Note: Since the Gutenberg SDK is still being actively developed, the developmen
     ```
 
-1.  In Jetpack, enable loading block assets by adding following filters to your local mu-plugins folder:
+1.  In Jetpack, enable loading local block assets by adding following filter to your local mu-plugins folder:
````
Can't we drop this line (and the other `add_filter()` statement) too, since we're now defaulting to `true` [here](https://github.com/Automattic/jetpack/pull/10292/files#diff-9ba9fa5bb52bd6cf36c6c0bdd5ed4830R7323)?

sirreal replied:
I misinterpreted this and indeed, CDN being disabled by default makes this bit of the doc unnecessary.

I've added more cleanup to the doc as part of this PR.

jeherve reviewed (dismissed):
This looks good to me, although I agree that the documentation is a bit confusing right now since it makes it sound like the filter must be used, when in fact it is not necessary anymore since the assets are now served locally by default.

tyxla commented on class.jetpack.php:
```diff
@@ -7302,17 +7296,18 @@ public static function is_gutenberg_available() {
+		}
+
 		/**
 		 * Filter to turn on loading Gutenberg blocks
```
Maybe change this to `Filter to turn off` now that it's on by default?

sirreal committed:
Update filter docblock

sirreal committed:
Update default filter value

sirreal committed:
Update docs

sirreal committed:
Remove backtick code-quotes from Jetpack

sirreal commented:
I've rebased and made some documentation changes as suggested by @tyxla and @ockham, this should be ready to land.

sirreal committed:
Remove docs about unmentioned filters

sirreal commented on docs/guides/gutenberg-blocks.md:
```diff
@@ -86,7 +58,22 @@ _Note: Since the Gutenberg SDK is still being actively developed, the developmen
+
+  If you use Jetpack-Docker, you could add these to `docker/wordpress/wp-config.php`
+
+  Those filters take action here: https://github.com/Automattic/jetpack/blob/b4a057fad975f3db8097fd62e702e276fd3d4389/class.jetpack.php#L7272-L7381
```
I just missed this when the section became optional and I moved it.

This doc is for getting Gutenbergers (📰 🍔) up and running. I think filters is beyond scope. Removed 🔪

ockham reviewed (dismissed):
:shipit:

jeherve commented on class.jetpack.php:
```diff
@@ -7302,17 +7296,18 @@ public static function is_gutenberg_available() {
 	 * @return void
 	 */
 	public static function load_jetpack_gutenberg() {
+		if ( ! Jetpack::is_active() ) {
```
Noting that while this works for now, we'll need to add more conditions here, as per #10198

jeherve commented on class.jetpack.php:
```diff
@@ -7274,14 +7274,8 @@ public static function is_gutenberg_available() {
+	 * Loading blocks is enabled by default and may be disabled via filter:
+	 *   add_filter( 'jetpack_gutenberg', '__return_false' );
 	 *
 	 * When enabled, blocks are loaded from CDN by default. To load locally instead:
```
Shouldn't this say "not loaded from CDN" by default now?

sirreal replied:
👍 Thanks for highlighting this.

I've removed the comment, I don't think it makes sense at this level. The filter is documented a bit lower. The CDN is no longer enabled by default and there doesn't seem to be much reason to enable it.

jeherve commented on docs/guides/gutenberg-blocks.md:
````diff
@@ -86,7 +58,20 @@ _Note: Since the Gutenberg SDK is still being actively developed, the developmen
+  define( 'GUTENBERG_DEVELOPMENT_MODE', true );
+  ```
+
+  If you use Jetpack-Docker, you could add these to `docker/wordpress/wp-config.php`
````
In other places of our documentation, we recommend using an mu-plugin instead:
https://github.com/Automattic/jetpack/tree/master/docker#must-use-plugins-directory

Should we recommend the same here?

sirreal replied:
We found that `SCRIPT_DEBUG` needs to be set earlier, which is why we're recommending `wp-config.php` here.

p1536745828000100-slack-luna

jeherve reviewed (changes requested):
I'd have a few questions before we merge this.

sirreal committed:
Remove docblock mentioning CDN

jeherve reviewed (approved):
Looking good now. 🚢
